README-trust.md

```diff
@@ -0,0 +1,116 @@
+Trust module
+============
+
+Description
+-----------
+
+The trust module allows to ensure presence and absence of a domain trust.
+
+Features
+--------
+* Trust management
+
+
+Supported FreeIPA Versions
+--------------------------
+
+FreeIPA versions 4.4.0 and up are supported by the ipagroup module.
+
+
+Requirements
+------------
+
+**Controller**
+* Ansible version: 2.8+
+
+**Node**
+* Supported FreeIPA version (see above)
+
+
+Usage
+=====
+
+Example inventory file
+
+```ini
+[ipaserver]
+ipaserver.test.local
+```
+
+
+Example playbook to create a one-way trust:
+
+```yaml
+---
+- name: Playbook to create a trust
+  hosts: ipaserver
+  become: true
+
+  tasks:
+  - name: create a one-way trust to the windows.local domain
+    ipatrust:
+      realm: windows.local
+      realm_admin: Administrator
+      realm_passwd: secret_password
+      state: present
+```
+
+Example playbook to create a two-way trust using a shared-secret:
+
+```yaml
+---
+- name: Playbook to create a trust
+  hosts: ipaserver
+  become: true
+
+  tasks:
+  - name: create a two-way trust to the windows.local domain
+    ipatrust:
+      realm: windows.local
+      trust_secret: my_share_Secret
+      biderectional: True
+      state: present
+```
+Example playbook to delete a trust:
+
+```yaml
+---
+- name: Playbook to delete trust
+  hosts: ipaserver
+  become: true
+
+  tasks:
+  - name: delete the trust with windows.local
+    ipatrust:
+      realm: windows.local
+      state: absent
+```
+This will only delete the ipa-side of the trust and it does NOT delete the id-range that matches the trust,  
+
+Variables
+=========
+
+ipatrust
+-------
+
+Variable | Description | Required
+-------- | ----------- | --------
+`ipaadmin_principal` | The admin principal is a string and defaults to `admin` | no
+`ipaadmin_password` | The admin password is a string and is required if there is no admin ticket available on the node | no
+`realm` | The realm name string. | yes
+`realm_admin` | Active Directory domain administrator string. | no
+`realm_passwd` | Active Directory domain administrator's password string. | no
+`realm_Server` | Domain controller for the Active Directory domain string. | no
+`trust_secret` | Shared secret for the trust string. | no
+`base_id` | First posix id for the trusted domain integer. | no
+`range_size` | Size of the ID range reserved for the trusted domain integer. | no
+`range_type` | Type of trusted domain ID range, It can be one of `ipa-ad-trust` or `ipa-ad-trust-posix`and defaults to `ipa-ad-trust`. | no
+`bidirectional` | Establish bi-directional trust. By default trust is inbound one-way only. (bool) | no
+`external` | Establish external trust to a domain in another forest. The trust is not transitive beyond the domain. (bool) | no
+`state` | The state to ensure. It can be one of `present` or `absent`, default: `present`. | yes
```

Comment:
Shouldn't the states `enable` and `disable` be available through trustdomain-enable/disable?

Reply:
The ipa trust-add command can only creat a trust that is always enabled.
The ipa trust-del command can only delete a trust.
However the libipa trust.py plugin also has code related to the 'ipa trustdomain-???' commands.
But these are not mentioned in the manuals from red hat (windows integration guide) and the howto's found on the freeipa.org domain on how to setup a trust.

I think they should be applied in a differen module ipatrustdomain since they are also a different command in the ipa cli.

Rob